instrumentation-genai/opentelemetry-instrumentation-google-genai/tests/generate_content/test_tool_call_instrumentation.py

```diff
@@ -0,0 +1,255 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+import google.genai.types as genai_types
+
+from .base import TestCase
+
+
+class ToolCallInstrumentationTestCase(TestCase):
+    def test_tool_calls_with_config_dict_outputs_spans(self):
+        calls = []
+
+        def handle(*args, **kwargs):
+            calls.append((args, kwargs))
+            return "some result"
+
+        def somefunction(somearg):
+            print("somearg=%s", somearg)
+
+        self.mock_generate_content.side_effect = handle
+        self.client.models.generate_content(
+            model="some-model-name",
+            contents="Some content",
+            config={
+                "tools": [somefunction],
+            },
+        )
+        self.assertEqual(len(calls), 1)
+        config = calls[0][1]["config"]
+        tools = config.tools
+        wrapped_somefunction = tools[0]
+
+        self.assertIsNone(self.otel.get_span_named("tool_call somefunction"))
+        wrapped_somefunction(somearg="someparam")
+        self.otel.assert_has_span_named("tool_call somefunction")
+        generated_span = self.otel.get_span_named("tool_call somefunction")
+        self.assertEqual(
+            generated_span.attributes["code.function.name"], "somefunction"
+        )
+
+    def test_tool_calls_with_config_object_outputs_spans(self):
+        calls = []
+
+        def handle(*args, **kwargs):
+            calls.append((args, kwargs))
+            return "some result"
+
+        def somefunction(somearg):
+            print("somearg=%s", somearg)
+
+        self.mock_generate_content.side_effect = handle
+        self.client.models.generate_content(
+            model="some-model-name",
+            contents="Some content",
+            config=genai_types.GenerateContentConfig(
+                tools=[somefunction],
+            ),
+        )
+        self.assertEqual(len(calls), 1)
+        config = calls[0][1]["config"]
+        tools = config.tools
+        wrapped_somefunction = tools[0]
+
+        self.assertIsNone(self.otel.get_span_named("tool_call somefunction"))
+        wrapped_somefunction(somearg="someparam")
+        self.otel.assert_has_span_named("tool_call somefunction")
+        generated_span = self.otel.get_span_named("tool_call somefunction")
+        self.assertEqual(
+            generated_span.attributes["code.function.name"], "somefunction"
+        )
+
+    def test_tool_calls_record_parameter_values_on_span_if_enabled(self):
+        os.environ["OTEL_INSTRUMENTATION_GENAI_CAPTURE_MESSAGE_CONTENT"] = (
```

Comment:
Can you update to `patch.dict` ([example](https://github.com/open-telemetry/opentelemetry-python/blob/71ec85930d61a1bc19d07ebe046f502882331d14/opentelemetry-api/tests/context/test_context.py#L87)) to isolate this test

Reply:
Done. Updated these tests to use `patch.dict`.